components/ReportSwitch.vue

```diff
@@ -18,17 +18,22 @@ const props = defineProps({
   },
 });
 const { report, toggleEulaPackageApproval } = useSingleReport();
-const currentlyActive = (eula) => report.value.state?.[props.name]?.[eula];
 
-const toggled = ref(currentlyActive(props.criterion));
+const toggled = ref(report.value.state?.[props.name]?.[props.criterion]);
+
+watchEffect(() => {
+  toggled.value = report.value.state?.[props.name]?.[props.criterion];
+});
```

Comment:
I cleaned up to use v-model again, let me know your thoughts

Reply:
This change throws a warning, unfortunately ... 

<img width="1624" alt="Screen Shot 2022-12-02 at 2 41 51 PM" src="https://user-images.githubusercontent.com/1477362/205308417-3138b85c-dd58-443f-9894-f6539eae7b41.png">

...and that warning opened a little saga.

At first I thought it appeared just because we were using a `computed` value for a `v-model`, which we should avoid because they don't have setters and conceptually are reserved for reactive values, with `data` or `ref` reserved for values we change. 

However, on actually reading the error, I see it is because the value we are using for `report` is being returned from `useQuery` and isn't writable for whatever reason is internal to the library. If we fix the error, by severing the object that is returned as `report` in `components/ReportSwitch.vue`, then the rest of the values in `composables/useSingleReport.js` that rely on `report` being globally changed will fail. 

When I try to sever the value from that being returned by `useQuery` in `composables/useSingleReport.js` (by doing something like `const report = Object.assign({}, data);` it creates an error that I don't fully understand — nothing is returned. 

I also notice that we are getting at least a one-second or so lag in updating the list when we toggle now that does not exist on `main`:

![lag](https://user-images.githubusercontent.com/1477362/205310932-f4eb2d12-0044-4458-8290-1e49ba5c790f.gif)

I believe that is because we are waiting for the toggle to write to the database before updating the list, which makes a bit of confusing UX. I think the ideal solution for that would be to 

All of this is to say, it appears to me that the one solution would be to:

1. Change `computed` to `ref`
2. Sever the object in  `components/ReportSwitch.vue`
3. Change the `toggleEulaPackageApproval` to take the entire report as an argument
4. Create a new issue for the timing, which based on what we have would likely need to be changed by emitting an event and then re-set if the DB failed, which would be a bit complex here.

A different opportunity here would be to radically simplify the way we are using state and updating the database. I've been thinking about this a little bit and originally was thinking it would be better to approach later (if at all), but I think maybe it would save us time in the end?

For that we would go old school and:
- create a shared data-structure in the parent component that feeds both the list and the detail view
- update it using events (Vue's `emit` is so nice for this) and then trigger the database calls at the top
  - basically update data structure optimistically, run db call, reset values on error
- pass the relevant data down as props

Fundamentally, this is a pretty simple app and I think a restructure might save us a lot of time and make it easier to accomplish the things we would want to with error handling, etc. If it ever got so large we needed more complex data-handling, we would then have a clearer idea of what needed that kind of handling. What do you think? Too much time? Worth it in the long run?